sound/soc/sof/intel/hda-codec.c

```diff
@@ -29,10 +29,15 @@ static void hda_codec_load_module(struct hda_codec *codec)
 {
 	char alias[MODULE_NAME_LEN];
 	const char *module = alias;
+	int res;
```

Comment:
I find the commit message rather confusing.

The codec probe should not be run just on a module load, this should be dependent on some event or a match with a specific ID, no?

Also I don't see the link with the machine driver, this is a patch against the platform driver? the failure may happen at the machine driver probe/init level but as a result of something missing earlier.

Reply:
@plbossart Yes, I'll work on the commit message. 
The legacy HDA driver has similar logic around request_module(). It seems the codec probe is not automatic, but has to be triggered like this. I'll dig a bit deeper, but definitely this is a major gap now in SOF, the probe does not get run in cases such as this.

And this is basicly a separate step of the platform driver. Platform driver needs to discover the HDA codecs present and load the correct driver modules.  Then when machine driver runs, then all the required entities are present and DAI links can be set up. Now we fail in machine driver snd_soc_register_card().